### PR TITLE
URO-41: Add KBaseButton component.

### PR DESCRIPTION
**Pull request by dakotablair**
I am opening this PR early to discuss the general approach. There are no unit tests yet, so they will need to be added before this PR may be merged.

### CODE REVIEW

File: src/common/KBaseButton.module.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
+.button {
+  /*button styles*/
+  border-color: #000;
```
Comment (dauglyon) [inline src/common/KBaseButton.module.scss:3]: Probably should use a kbase color from colors.scss (here and below)

**Review by dauglyon — COMMENTED** [review]
Overall this looks good, though "KBaseButton" is a long name. Maybe we just call it "Button"? Also added some inline comments.

File: src/common/KBaseButton.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  children,
+  disabled,
+  onclick,
+}) => {
+  return (
+    <button className={classes.button} onClick={onclick} disabled={disabled}>
+      {children}
+    </button>
+  );
+};
```
Comment (dauglyon) [inline src/common/KBaseButton.tsx:19]:
```suggestion
export const KBaseButton: FC<React.HTMLProps<HTMLButtonElement>> = (props) => {
  return (
    <button {...props} className={[classes.button, props.className].join(' ')}>
      {props.children}
    </button>
  );
};
```

File: src/common/KBaseButton.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { FunctionComponent as FC, MouseEventHandler } from 'react';
+import classes from './KBaseButton.module.scss';
+
+interface KBaseComponentProps {
+  disabled?: boolean;
+  onclick?: MouseEventHandler;
+}
```
Comment (dauglyon) [inline src/common/KBaseButton.tsx:7]:
```suggestion
```

**Review by dauglyon — COMMENTED** [review]
Also, the button props could just be recycled from `React.HTMLProps<HTMLButtonElement>` and then spread into the button. Custom props could be added later with `interface KbaseButtonProps extends React.HTMLProps<HTMLButtonElement> {someCustomeProp:type;}`

File: src/common/KBaseButton.module.scss (hunk 1 above)
Comment (dakotablair) [reply src/common/KBaseButton.module.scss]: I might have gone too far in the other direction, but I think we'll want [classes like this](https://github.com/kbase/ui-refresh-test/pull/15/files#diff-6ea08c95a60c10153bc1d5ae1e330e1fa103cc61df85954a91f741359450a5b0R28) at some point anyway. Let me know what you think.

File: src/common/KBaseButton.tsx (hunk 2 above)
Comment (dakotablair) [reply src/common/KBaseButton.tsx]: This was almost perfect advice, but the type ended up being `FC<React.HTMLAttributes<HTMLButtonElement>>` after running into [this issue](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/36505).

File: src/common/components/Button.module.scss [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  &[disabled] {
+    /*disabled button styles*/
+    background-color: use_color("neutral");
+    border-color: use_color("neutral");
+    color: use_color("silver");
+  }
+}
+
+@each $state, $_ in $palette {
```
Comment (dauglyon) [inline src/common/components/Button.module.scss:15]: scss power ftw

**Review by dauglyon — APPROVED** [review]
lgtm!

**Pull request merged**